pkg/controller/garbagecollector/operations.go

```diff
@@ -34,7 +34,7 @@ import (
 // namespace> tuple to a unversioned.APIResource struct.
 func (gc *GarbageCollector) apiResource(apiVersion, kind string) (*metav1.APIResource, error) {
 	fqKind := schema.FromAPIVersionAndKind(apiVersion, kind)
-	mapping, err := gc.restMapper.RESTMapping(fqKind.GroupKind(), apiVersion)
+	mapping, err := gc.restMapper.RESTMapping(fqKind.GroupKind(), fqKind.Version)
```

Comment:
I *think* the fact that GC is currently working, when we were passing in an invalid version here means that the current behavior of PriorityRESTMapper treats the specified versions as a hint, not a requirement.

Fixing this bug here to pass the right version is good, but I'm not as sure about the change at https://github.com/kubernetes/kubernetes/pull/55218/files#diff-1054b3cbb6002ca622530dd7b2733558R156